src/sortStudents.ts

```diff
@@ -1,16 +1,56 @@
 
 export interface Student {
-  // describe Student interface
+  name: string;
+  surname: string;
+  age: number;
+  married: boolean;
+  grades: number[];
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name = 'name',
+  Surname = 'surname',
+  Age = 'age',
+  Married = 'married',
+  AverageGrade = 'grades',
 }
 
 // create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
 
+function averageGrades(grades: number[]): number {
+  return grades.reduce((sum, grade) => sum + grade, 0) / grades.length;
+}
+
+export function sortStudents(students: Student[], sortBy: SortType,
+  order: SortOrder)
+  : Student[] {
+  const copyStudent: Student[] = [...students];
 
-export function sortStudents(students, sortBy, order) {
-  // write your function
+  switch (sortBy) {
+    case SortType.Name:
+    case SortType.Surname:
+      return order === 'asc'
+        ? copyStudent.sort((student, nextStudent) => student[sortBy]
+          .localeCompare(nextStudent[sortBy]))
+        : copyStudent.sort((student, nextStudent) => nextStudent[sortBy]
+          .localeCompare(student[sortBy]));
+    case SortType.Age:
```

Comment:
Between cases you can add blank line

```suggestion
          .localeCompare(student[sortBy]));

    case SortType.Age:
```